lib/dns/dns.go

```diff
@@ -639,3 +640,94 @@ func (s *Server) Stop() {
 func (s *Server) Wait() {
 	s.wg.Wait()
 }
+
+type FQDNs map[string]bool
```

Comment:
nitpick, can we name this something other than 'FQDNs'?  It's not clear what the bool is for unless the type name is clear.  It seems the bool is only used by the middleware.  If it's not needed by the dns package, I'd rather we return an array of string and then have the main.go do the necessary work to make it work for the middleware.

Reply:
I'm using a map in the absence of sets in Go so that if there are a lot of domains we don't have to iterate through the array (notice the checking routine lacks a loop!)

I can name this a little better so that it's more clear, though.